driver/web/transaction.go

```diff
@@ -53,112 +42,41 @@ func (t *Transaction) Config() (config transaction.Config) {
 
 // Run executes the web transaction
 func (t *Transaction) Run(ctx context.Context) (result transaction.Result) {
+	t.driver.m.RLock()
+	defer t.driver.m.RUnlock()
 
 	var err error
-
-	if t.timeout > 0 {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeout(ctx, t.timeout)
-		defer cancel()
-	}
-
 	result.Name = t.config.Name
 	result.Driver = "web"
-	result.Time = time.Now()
 	result.Metadata = t.config.Metadata
 
-	cdpAddr, err := t.resolveAddr()
-	if err != nil {
-		result.Error = fmt.Errorf("driver/web: %w", err)
-		result.Message = "failed to handle cdp url"
-		result.DurationSeconds = time.Since(result.Time).Seconds()
-		return result
-	}
-
-	switch t.proxied {
+	// handle browserless mode for testing
+	var drv *cdp.Driver
+	switch t.driver.config.testing {
```

Comment:
Isn't this more readable with if else?

Reply:
This is a matter of taste. I prefer the switch statement which make the cases explicit and more "readable".